rpcd/playbooks/roles/elasticsearch/tasks/upgrade_elasticsearch_pre.yml

```diff
@@ -0,0 +1,256 @@
+  with_items: elasticsearch_apt_packages
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install pip packages for upgrade
+  pip:
+    name: "{{ item }}"
+    state: latest
+    extra_args: "{{ pip_install_options | default('') }}"
+  register: install_pip_packages
+  until: install_pip_packages|success
+  retries: 5
+  delay: 2
+  with_items: elasticsearch_upgrade_pip_packages
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Stop elasticsearch
+  service:
+    name: elasticsearch
+    state: stopped
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Make sure elasticsearch is stopped
+  command: 'pkill -f elasticsearch'
+  ignore_errors: true
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Start elasticsearch
+  service:
+    name: elasticsearch
+    state: started
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Wait for elasticsearch to restart
+  wait_for:
+    host: "{{ hostvars[groups['elasticsearch_container'][0]]['container_address'] }}"
+    port: 9200
+    state: started
+    timeout: 300
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Drop log_test1 template
+  shell: 'curl -sk -XDELETE {{ internal_lb_vip_address }}:{{ elasticsearch_http_port }}/_template/log_test1'
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+    - skip_ansible_lint
+
+- name: Place legacy liberty template script
+  template:
+    src: legacy-mapping.sh.j2
+    dest: /opt/legacy-mapping.sh
+    owner: root
+    group: root
+    mode: 0755
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install legacy liberty template
+  shell: '/opt/legacy-mapping.sh'
+  register: legacy_template_result
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Remove legacy liberty template script
+  file:
+    name: '/opt/legacy-mapping.sh'
+    state: absent
+  delegate_to: "{{ groups['elasticsearch_all'][0] }}"
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Install Reindexing Plugin
+  command: ./plugin -install {{ item }}
+  args:
+    chdir: /usr/share/elasticsearch/bin
+    creates: /usr/share/elasticsearch/plugins/reindexing
+  with_items:
+    - org.codelibs/elasticsearch-reindexing/1.7.0
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Stop elasticsearch
+  service:
+    name: elasticsearch
+    state: stopped
+  tags:
+    - logging-upgrade
+    - elasticsearch-upgrade
+    - reindex-wrapper
+
+- name: Make sure elasticsearch is stopped
+  command: 'pkill -f elasticsearch'
```

Comment:
Ideally, that's what I would do, in the interest of time, however, this works; but I agree it's terrible.